### PR TITLE
block nil validation

### PR DESCRIPTION
**Pull request by ucwong**

### CODE REVIEW

File: eth/api_backend.go [hunk 1]
```diff
@@ -67,11 +67,18 @@ func (b *EthAPIBackend) HeaderByNumber(ctx context.Context, number rpc.BlockNumb
...
-		return block.Header(), nil
+		if block != nil {
+			return block.Header(), nil
+		}
+		return nil, errors.New("pending block not found")
 	}
 	// Otherwise resolve and return the block
 	if number == rpc.LatestBlockNumber {
-		return b.eth.blockchain.CurrentBlock().Header(), nil
+		block := b.eth.blockchain.CurrentBlock()
```
Comment (karalabe) [inline eth/api_backend.go:77]: The current block can never be nil.

File: eth/api_backend.go [hunk 2]
```diff
@@ -67,11 +67,18 @@ func (b *EthAPIBackend) HeaderByNumber(ctx context.Context, number rpc.BlockNumb
 	// Pending block is only known by the miner
 	if number == rpc.PendingBlockNumber {
 		block := b.eth.miner.PendingBlock()
-		return block.Header(), nil
+		if block != nil {
```
Comment (karalabe) [inline eth/api_backend.go:70]: The pending block - if I understand it correctly - can only be empty on startup, but will then be filled and will always have value afterwards.

**Comment by karalabe** [comment]
Unless I'm missing something, these cannot be `nil`s, so no need for the defensive checks.

**Review by tuanggolt — COMMENTED** [review]
open

**Pull request closed without merging**